src/transactions/test/TxEnvelopeTests.cpp

```diff
@@ -2628,6 +2628,57 @@ TEST_CASE("soroban transaction validation", "[tx][envelope][soroban]")
             REQUIRE(!tx->checkValid(*app, ltx, 0, 0, 0));
             REQUIRE(tx->getResult().result.code() == txSOROBAN_INVALID);
         }
+        SECTION("resource fee is negative")
+        {
+            auto tx = sorobanTransactionFrameFromOpsWithTotalFee(
+                app->getNetworkID(), root, {op0}, {}, resources, 1'000'000,
+                std::numeric_limits<int64_t>::min());
+            LedgerTxn ltx(app->getLedgerTxnRoot());
+            REQUIRE(!tx->checkValid(*app, ltx, 0, 0, 0));
+            // Negative resource fee is handled before we get to
+            // Soroban-specific checks.
+            REQUIRE(tx->getResult().result.code() == txMALFORMED);
+        }
+        SECTION("resource fee exceeds uint32")
+        {
+            auto tx = sorobanTransactionFrameFromOpsWithTotalFee(
+                app->getNetworkID(), root, {op0}, {}, resources,
+                std::numeric_limits<uint32_t>::max(),
+                std::numeric_limits<int64_t>::max());
+            LedgerTxn ltx(app->getLedgerTxnRoot());
+            REQUIRE(!tx->checkValid(*app, ltx, 0, 0, 0));
+            REQUIRE(tx->getResult().result.code() == txSOROBAN_INVALID);
+        }
+        SECTION("resource fee exceeds uint32 after adding base fee")
+        {
+            auto tx = sorobanTransactionFrameFromOpsWithTotalFee(
+                app->getNetworkID(), root, {op0}, {}, resources,
+                std::numeric_limits<uint32_t>::max(),
+                static_cast<int64_t>(std::numeric_limits<uint32_t>::max()) -
+                    100 + 1);
+            LedgerTxn ltx(app->getLedgerTxnRoot());
+            REQUIRE(!tx->checkValid(*app, ltx, 0, 0, 0));
+            // This gets rejected due to insufficient inclusion fee, so
```

Comment:
does it make sense to also cover the case where there is enough balance to cover full fee, but resource fee exceeds uint32 after adding base fee?

Reply:
Not sure I understand your setup; it seems like you suggest what we're already doing in this scenario. I've added a successful case though.